plugin/discovery/get.go

```diff
@@ -93,6 +104,18 @@ func (i *ProviderInstaller) Get(provider string, req Constraints) (PluginMeta, e
 	// take the first matching plugin we find
 	for _, v := range versions {
 		url := providerURL(provider, v.String())
+
+		if !i.SkipVerify {
+			sha256, err := getProviderChecksum(provider, v.String())
+			if err != nil {
+				return PluginMeta{}, nil
+			}
+
+			if sha256 != "" {
+				url = url + "?checksum=sha256:" + sha256
```

Comment:
This is a go-getter feature, it's "API" is via the URL passed in. It will return an error if the file doesn't match the checksum provided. I may still want to augment or drop go-getter for plugin downloads, since I think we should also verify that the zip file is unpacking into the expected filename, otherwise we will just keep fetching the same thing and failing.